src/main/java/frc/trigon/robot/subsystems/arm/Arm.java

```diff
@@ -0,0 +1,201 @@
+                    getSetTargetElevatorPositionCommand(targetElevatorPosition, elevatorSpeedPercentage)
+            );
+        }
+        return new SequentialCommandGroup(
+                getSetTargetElevatorPositionCommand(targetElevatorPosition, elevatorSpeedPercentage),
+                getSetTargetAngleCommand(targetAngle, angleSpeedPercentage)
+        );
+    }
+
+    private Command getSetTargetAngleCommand(Rotation2d targetAngle, double speedPercentage) {
+        return new FunctionalCommand(
+                () -> generateAngleMotorProfile(targetAngle, speedPercentage),
+                this::setTargetAngleFromProfile,
+                (interrupted) -> {
+                },
+                () -> false,
+                this
+        );
+    }
+
+    private Command getSetTargetElevatorPositionCommand(double targetElevatorPosition, double speedPercentage) {
+        return new FunctionalCommand(
+                () -> generateElevatorMotorProfile(targetElevatorPosition, speedPercentage),
+                this::setTargetElevatorPositionFromProfile,
+                (interrupted) -> {
+                },
+                () -> false,
+                this
+        );
+    }
+
+    private void generateAngleMotorProfile(Rotation2d targetAngle, double speedPercentage) {
+        angleMotorProfile = new TrapezoidProfile(
+                Conversions.scaleConstraints(ArmConstants.ANGLE_CONSTRAINTS, speedPercentage),
+                new TrapezoidProfile.State(targetAngle.getDegrees(), 0),
+                new TrapezoidProfile.State(getAnglePositionDegrees().getDegrees(), getAngleVelocityDegreesPerSecond())
+        );
+        lastAngleMotorProfileGenerationTime = Timer.getFPGATimestamp();
+    }
+
+    private void generateElevatorMotorProfile(double targetElevatorPosition, double speedPercentage) {
+        elevatorMotorProfile = new TrapezoidProfile(
+                Conversions.scaleConstraints(ArmConstants.ELEVATOR_CONSTRAINTS, speedPercentage),
+                new TrapezoidProfile.State(targetElevatorPosition, 0),
+                new TrapezoidProfile.State(getElevatorPositionRevolutions(), getElevatorVelocityRevolutionsPerSecond())
+        );
+        lastElevatorMotorProfileGenerationTime = Timer.getFPGATimestamp();
+    }
+
+    private void setTargetAngleFromProfile() {
+        if (angleMotorProfile == null) {
+            stopAngleMotors();
+            return;
+        }
+
+        TrapezoidProfile.State targetState = angleMotorProfile.calculate(getAngleProfileTime());
+        armIO.setAngleMotorPower((calculateAngleMotorVoltage(targetState)));
+    }
+
+    private void setTargetElevatorPositionFromProfile() {
+        if (elevatorMotorProfile == null) {
+            stopElevatorMotors();
+            return;
+        }
+
+        TrapezoidProfile.State targetState = elevatorMotorProfile.calculate(getElevatorProfileTime());
+        armIO.setElevatorMotorPower((calculateElevatorMotorVoltage(targetState)));
```

Comment:
use a `setTargetElevatorPosition` function here that should be a part of the IO.